tests/integration/test_bundle.py

```diff
@@ -276,6 +278,35 @@ async def test_loki_receives_logs(ops_test: OpsTest):
     logger.info("loki is successfully receiving flog logs")
 
 
+@pytest.mark.abort_on_fail
+async def test_all_traefik_proxied_urls_return_200(ops_test: OpsTest):
+    """Verify that all ingressed applications' URLs are reachable from localhost and return 200."""
+    proxied_urls = await get_all_proxied_urls(ops_test)
+    for app, url in proxied_urls.items():
+        response = urlopen(url, context=insecure_context)
+        assert response.code == 200
```

Comment:
For easier failure troubleshooting, perhaps we could add the app and url to the assert message?
```suggestion
        assert response.code == 200, f"Unreachable: code={response.code} app={app}, url={url}"
```

Reply:
Not sure what urlopen does with redirect codes (30x).
Should the assertion be changed to `<400`?